bioptim/dynamics/problem.py

```diff
@@ -33,8 +34,105 @@ def torque_driven(ocp, nlp):
         else:
             Problem.configure_forward_dyn_func(ocp, nlp, DynamicsFunctions.forward_dynamics_torque_driven)
 
+    @staticmethod
+    def torque_driven_with_actuator_constraints(ocp, nlp):
+        """
+        Names states (nlp.x) and controls (nlp.u) and gives size to (nlp.nx) and (nlp.nu).
+        Works with torques but without muscles, must be used with dynamics without contacts.
+        :param nlp: An instance of the OptimalControlProgram class.
+        """
+        def tau_actuator_constraints(ocp, nlp, t, x, u, p, minimal_tau=None):
+            nq = nlp.mapping["q"].reduce.len
+            q = [nlp.mapping["q"].expand.map(mx[:nq]) for mx in x]
+            q_dot = [nlp.mapping["q_dot"].expand.map(mx[nq:]) for mx in x]
+
+            min_bound = []
+            max_bound = []
+
+            func = biorbd.to_casadi_func("torqueMax", nlp.model.torqueMax, nlp.q, nlp.q_dot)
+            for i in range(len(u)):
+                bound = func(q[i], q_dot[i])
+                if minimal_tau:
+                    min_bound.append(nlp.mapping["tau"].reduce.map(if_else(lt(bound[:, 1], minimal_tau), minimal_tau, bound[:, 1])))
+                    max_bound.append(nlp.mapping["tau"].reduce.map(if_else(lt(bound[:, 0], minimal_tau), minimal_tau, bound[:, 0])))
+                else:
+                    min_bound.append(nlp.mapping["tau"].reduce.map(bound[:, 1]))
+                    max_bound.append(nlp.mapping["tau"].reduce.map(bound[:, 0]))
+
+            obj = vertcat(*u)
+            min_bound = vertcat(*min_bound)
+            max_bound = vertcat(*max_bound)
+
+            return (
+                vertcat(np.zeros(min_bound.shape), np.ones(max_bound.shape) * -np.inf),
+                vertcat(obj + min_bound, obj - max_bound),
+                vertcat(np.ones(min_bound.shape) * np.inf, np.zeros(max_bound.shape)),
+            )
+
+        constraints = ConstraintList()
+        for i in range(ocp.nb_phases):
+            constraints.add(tau_actuator_constraints, phase=i, node=Node.ALL)
+        ocp.update_constraints(constraints)
```

Comment:
I feel (it has to be confirmed) that putting and update_constraints here has the disadvantage of messing up the g vector (by putting constraints at the beginning instead of after the continuity constraint). This may make this hard to warm start by moving node afterwards..
I don't see good solution with that apart from adding a flag that would be interpreted by optimal_control_program to call "pending constraints"... Would that make sense?